### PR TITLE
Add support for builtin/builtin routine 'name' property

### PR DESCRIPTION
**Pull request by rerobika**
JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtins.c [hunk 1]
```diff
@@ -608,7 +614,11 @@ ecma_builtin_make_function_object_for_setter_accessor (ecma_builtin_id_t builtin
                                                                             *   object's routine property */
                                                        uint16_t name_id) /**< magic string id of 'name' property */
 {
-  return ecma_builtin_make_function_object_for_routine (builtin_id, routine_id, name_id, 1);
+  return ecma_builtin_make_function_object_for_routine (builtin_id,
+                                                        routine_id,
+                                                        name_id,
+                                                        ECMA_BUILTIN_ROUTINE_GETTER,
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c:620]: ecma_builtin_make_function_object_for_setter_accessor: ECMA_BUILTIN_ROUTINE_GETTER???

File: jerry-core/ecma/builtin-objects/ecma-builtins.c [hunk 2]
```diff
@@ -667,6 +677,67 @@ ecma_builtin_routine_try_to_instantiate_property (ecma_object_t *object_p, /**<
...
+#if ENABLED (JERRY_ES2015)
+  /*
+   * Lazy instantiation of 'name' property
+   */
+  if (ecma_compare_ecma_string_to_magic_id (string_p, LIT_MAGIC_STRING_NAME))
+  {
+    ecma_property_t *name_prop_p;
+    ecma_extended_object_t *ext_func_p = (ecma_extended_object_t *) object_p;
+    uint16_t *bitset_p = &ext_func_p->u.built_in.u.builtin_routine.bitset;
+    if (*bitset_p & ECMA_BUILTIN_ROUTINE_NAME_INITIALIZED)
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c:689]: Newline before

File: jerry-core/ecma/builtin-objects/ecma-builtins.c [hunk 3]
```diff
@@ -667,6 +677,67 @@ ecma_builtin_routine_try_to_instantiate_property (ecma_object_t *object_p, /**<
...
+    {
+      /* name property was already instantiated */
+      return NULL;
+    }
+    /* We mark that the property was lazily instantiated */
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c:694]: There are other cases where newline should be added.

File: jerry-core/ecma/builtin-objects/ecma-builtins.c [hunk 4]
```diff
@@ -667,6 +677,67 @@ ecma_builtin_routine_try_to_instantiate_property (ecma_object_t *object_p, /**<
     return len_prop_p;
   }
 
+#if ENABLED (JERRY_ES2015)
+  /*
+   * Lazy instantiation of 'name' property
+   */
+  if (ecma_compare_ecma_string_to_magic_id (string_p, LIT_MAGIC_STRING_NAME))
+  {
+    ecma_property_t *name_prop_p;
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c:686]: This can be moved just before creating the property.

**Comment by zherczeg** [comment]
We should think about adding some basic tests. No need to cover everything, but one normal built-in function, a getter and a setter.

**Comment by rerobika** [comment]
@zherczeg @galpeter Thanks for the reviews, the PR is updated.

**Pull request merged**